### DISCUSSION TITLE
Reading Value Returned by Function during useContractWrite()

### DISCUSSION THREAD

**Discussion by 0xNotes** [wevm/wagmi#691] [Q&A]
Assume I have a Factory contract, the `generate()` function generates a new contract, it also returns the address of that contract. 
To execute this I would need to use `useContractWrite()`, how would I read the values in the returned `data` from this? 

A simple example would be below. 
How do I get the  `address` returned from using `useContractWrite()` on `generate()`?

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.10;


contract Factory {
      Child[] children;
      function generate(uint data) public returns (address){
         Child child = new Child(data);
         children.push(child);
         return(address(child));
      }
}

contract Child{
     uint data;
     constructor(uint _data){
        data = _data;
     }
}
```

**Comment by nasa8x** [comment]
```
import { useContractWrite } from 'wagmi'

function App() {
  const { data, isError, isLoading, write } = useContractWrite({
    addressOrName: 'your_contract_address',
    contractInterface: FactoryABI,
    functionName: 'generate',
    args: [1234],
  })

  return <button onClick={() => write()}> Generate </button>
}
```
`data` will return the address you want

**Reply by nasa8x** [reply]
Or use function to invoke when write is successful: <https://wagmi.sh/docs/hooks/useContractWrite>
```js 
onSuccess(data) {
      console.log('Success', data)
 },
```

**Comment by anteqkois** [comment]
@nasa8x response returns hash transaction, not deployed address :/
Anyone have idea how to solve it ? I don't want to use useContractRead, It would be better if I can get data in one transaction. I have to connect data from form and returned contract address (from transaction) and save in database, so any delay will be disadvantegous.
